packages/theme/getStore.ts

```diff
@@ -0,0 +1,58 @@
+import axios from 'axios';
+import urlJoin from 'url-join';
+
+interface StoreConfig {
+  backendUrl: string
+  storeApiRoute: string
+}
+
+/* eslint-disable camelcase */
+export interface StoreResponse {
+  data: {
+    type: string
+    id: string
+    attributes: {
+      name: string
+      url: string
+      meta_description: string | null
+      meta_keywords: string | null
+      seo_title: string | null
+      default_currency: string
+      supported_currencies: string
+      default_locale: string
+      supported_locales: string
+    }
+  }
+}
+interface Store {
+  name: string
+  url: string
+  defaultCurrency: string
+  supportedCurrencies: string
+  defaultLocale: string
+  supportedLocales: string
+  metaDescription?: string
+  metaKeywords?: string
+  seoTitle?: string
+}
+
+const deserializeStore = ({ data: { attributes } }: StoreResponse): Store => ({
+  name: attributes.name,
+  url: attributes.url,
+  metaDescription: attributes.meta_description,
+  metaKeywords: attributes.meta_keywords,
+  seoTitle: attributes.seo_title,
+  defaultCurrency: attributes.default_currency,
+  supportedCurrencies: attributes.supported_currencies,
+  defaultLocale: attributes.default_locale,
+  supportedLocales: attributes.supported_locales
+});
+
+const getStore = async (config: StoreConfig): Promise<Store> => {
+  const { data } = (
+    await axios.get(urlJoin(config.backendUrl, config.storeApiRoute))
+  ) as { data: StoreResponse };
```

Comment:
Actually, you could fulfill the generic type the [get](https://github.com/axios/axios/blob/7d6bddba2d8de29c263feaef4c40daa50cb4b176/index.d.ts#L146) method exposes.
```suggestion
  const { data } = await axios.get<StoreResponse>(urlJoin(config.backendUrl, config.storeApiRoute)
```

Additionally, for readability, I'd instead export the URL into an additional variable, so:
```suggestion
  const requestUrl = urlJoin(config.backendUrl, config.storeApiRoute)
  const { data } = await axios.get<StoreResponse>(requestUrl)
```

WDYT?

Reply:
Looks much cleaner, thanks!